src/validations/rules/ssp.sch

```diff
@@ -3450,9 +3450,66 @@
+        <sch:rule
+            context="oscal:control-implementation">
+            <sch:let
+                name="sensitivity-level"
+                value="/ => lv:sensitivity-level()" />
+            <sch:let
+                name="selected-profile"
+                value="$sensitivity-level => lv:profile()" />
+            <sch:let
+                name="required-controls"
+                value="$selected-profile/*//oscal:control/@id ! xs:string(.)" />
+            <sch:let
+                name="implemented-controls"
+                value="oscal:implemented-requirement/@control-id ! xs:string(.)" />
+            <sch:let
+                name="technical-controls"
+                value="$fedramp-values//fedramp:value-set[@name eq 'technical-control-id']//fedramp:enum/@value ! xs:string(.)" />
+            <sch:let
+                name="required-technical-controls"
+                value="$technical-controls[. = $required-controls]" />
+            <sch:let
+                name="missing-required-technical-controls"
+                value="$required-technical-controls[not(. = $implemented-controls)]" />
+            <sch:assert
+                diagnostics="technical-control-exists-diagnostic"
+                id="technical-control-exists"
+                role="error"
+                test="
+                    every $item in $missing-required-technical-controls
+                        satisfies exists(//oscal:implemented-requirement[@control-id eq $item])">Every required technical control is
+                implemented.</sch:assert>
+        </sch:rule>
         <sch:rule
             context="oscal:implemented-requirement"
             doc:template-reference="System Security Plan Template §13">
+            <sch:let
+                name="sensitivity-level"
+                value="/ => lv:sensitivity-level()" />
+            <sch:let
+                name="selected-profile"
+                value="$sensitivity-level => lv:profile()" />
+            <sch:let
+                name="technical-controls"
+                value="$fedramp-values//fedramp:value-set[@name eq 'technical-control-id']//fedramp:enum/@value ! xs:string(.)" />
+            <sch:assert
+                diagnostics="technical-control-is-implemented-diagnostic"
+                id="technical-control-is-implemented"
+                role="error"
+                test="
+                    if (@control-id = $technical-controls)
+                    then
+                        (
+                        if (exists(oscal:prop[@ns eq 'https://fedramp.gov/ns/oscal' and @name = 'implementation-status' and @value eq 'implemented']))
+                        then
+                            (true())
+                        else
+                            (false()))
+                    else
+                        (true())">The technical control-implementation <xsl:value-of
+                    select="@control-id" /> has a status of 'implemented'.</sch:assert>
```

Comment:
To be precise this should be "has an **implementation-status** of 'implemented'.".

The context is not "technical control-implementation" - it is a `implemented-requirement` and the assertion is testing that the technical control's `implemented-requirement` implementation status is 'implemented') The the assertion message should be something like "Every technical control is fully implemented." or "Technical control xx-x is implemented".

Reply:
fixed